analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.CSharp7.cs

```diff
@@ -196,9 +193,25 @@ public static void M(string path, int timeoutmilliseconds = 500)
                 {
                     System.Threading.Thread.Sleep(timeoutmilliseconds);
                 }
-            } while (!deletesuccess); // Compliant
+            } while (!deletesuccess);   // Noncompliant: This repro is written badly. Look below for a fixed version
```

Comment:
Let's delete this unfixed version of the test?
They seem identical - why it's not raising in the second example?

Reply:
I don't want to delete a repro referring to an existing issue.
The first example raises as expected. The assignment will never fail => the variable is always true.
The second example can fail on the call to `Console.WriteLine` and thus might not always set the variable to true.

Reply:
This comment needs updating IMO - it's not really clear what it means etc.

Reply:
Adding why one raises and the other doesn't would be helpful (for my eyes at least).
It was not trivial for me o understand that by adding the `Console.WriteLine` you immediately create the path of having a try block to fail.